prophet/prophet.go

```
@@ -0,0 +1,138 @@
+
+// Results returns a slice with all the results of futures that have been
+// executed.
+// The returned function must be called to remove the results from the set.
+func (p *P) Results() ([]FutureResult, func()) {
+	values := p.resultsWriter.Values()
+	if len(values) == 0 {
+		return nil, func() {}
+	}
+
+	return values, func() {
+		p.resultsWriter.Remove(values...)
```

golangci-lint (1.62.2)

Error return value of `p.resultsWriter.Remove` is not checked

(errcheck)